pytorch_lightning/core/lightning.py

```diff
@@ -1115,6 +1115,25 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: Optional[int]
         By default, it calls :meth:`~pytorch_lightning.core.lightning.LightningModule.forward`.
         Override to add any processing logic.
 
+        The :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` is used
+        to scale inference on multi-devices.
+
+        To prevent an OOM error, it is possible to use :class:`~pytorch_lightning.callbacks.BasePredictionWriter`
```

Comment:
Could also add, predictions are not returned for Spawn Training Plugins.

Reply:
@kaushikb11 I wonder if this is something we might want to change with the approach from #7916 .

Reply:
We could. It was why we introduced this :)

Reply:
Done !